api/src/test/java/org/pmiops/workbench/billing/BillingAlertsServiceTest.java

```diff
@@ -96,75 +122,246 @@ public void checkFreeTierBillingUsage_disabledUserNotIgnored() {
+    workbenchConfig.billing.defaultFreeCreditsLimit = 100.0;
+
+    User user = createUser("test@test.com");
+    createWorkspace(user, SINGLE_WORKSPACE_TEST_PROJECT, BillingMigrationStatus.OLD);
+    billingAlertsService.checkFreeTierBillingUsage();
+
+    verifyZeroInteractions(notificationService);
+    assertThat(workspaceFreeTierUsageDao.count()).isEqualTo(0);
+  }
+
+  @Test
+  public void checkFreeTierBillingUsage_ignoreMIGRATEDMigrationStatus() {
+    workbenchConfig.billing.defaultFreeCreditsLimit = 100.0;
+
+    User user = createUser("test@test.com");
+    createWorkspace(user, SINGLE_WORKSPACE_TEST_PROJECT, BillingMigrationStatus.MIGRATED);
+    billingAlertsService.checkFreeTierBillingUsage();
+
+    verifyZeroInteractions(notificationService);
+    assertThat(workspaceFreeTierUsageDao.count()).isEqualTo(0);
+  }
+
+  @Test
+  public void checkFreeTierBillingUsage_dbUpdate() {
+    workbenchConfig.billing.defaultFreeCreditsLimit = 100.0;
 
     User user = createUser("test@test.com");
-    createWorkspace(user, "aou-test-f1-26");
+    Workspace workspace = createWorkspace(user, SINGLE_WORKSPACE_TEST_PROJECT);
 
     billingAlertsService.checkFreeTierBillingUsage();
     verify(notificationService).alertUser(eq(user), any());
+    assertSingleWorkspaceTestDbState(user, workspace, BillingStatus.INACTIVE);
 
-    user.setFreeTierCreditsLimitOverride(1000.0);
-    userDao.save(user);
+    Timestamp t0 = workspaceFreeTierUsageDao.findAll().iterator().next().getLastUpdateTime();
+
+    // time elapses, and this project incurs additional cost
+
+    Map<String, Double> newResults = new HashMap<>();
+    newResults.put(SINGLE_WORKSPACE_TEST_PROJECT, SINGLE_WORKSPACE_TEST_COST * 2);
+    doReturn(mockTableResult(newResults)).when(bigQueryService).executeQuery(any());
+
+    // we alert again, and the cost field is updated in the DB
 
     billingAlertsService.checkFreeTierBillingUsage();
-    verifyZeroInteractions(notificationService);
+    verify(notificationService, times(2)).alertUser(eq(user), any());
+
+    // retrieve from DB again to reflect update after cron
+    Workspace dbWorkspace = workspaceDao.findOne(workspace.getWorkspaceId());
+    assertThat(dbWorkspace.getBillingStatus()).isEqualTo(BillingStatus.INACTIVE);
+    assertThat(dbWorkspace.getBillingAccountType()).isEqualTo(BillingAccountType.FREE_TIER);
+
+    assertThat(workspaceFreeTierUsageDao.count()).isEqualTo(1);
+    WorkspaceFreeTierUsage dbEntry = workspaceFreeTierUsageDao.findAll().iterator().next();
+    assertThat(dbEntry.getUserId()).isEqualTo(user.getUserId());
+    assertThat(dbEntry.getWorkspaceId()).isEqualTo(workspace.getWorkspaceId());
+    assertThat(dbEntry.getCost())
+        .isCloseTo(SINGLE_WORKSPACE_TEST_COST * 2, withinPercentage(0.001));
+
+    Timestamp t1 = dbEntry.getLastUpdateTime();
+    assertThat(t1).isAfter(t0);
+  }
+
+  private TableResult mockTableResult(Map<String, Double> costMap) {
```

Comment:
It took me way too long to figure out that this is mocking a BigQuery thing instead of a SQL thing. This is certainly because I'm not familiar with talking to BigQuery in Java, but it would still be helpful for me if this was called something like `mockBigQueryTableResult` instead.